### PR TITLE
[#1119] Decouple Actor from Collision System

### LINKED ISSUE #1119: [Refactor] Decouple the Collision system from Actor
### Context
Currently the colllision system has references to actor all throughout. This makes it difficult to make changes and test the collision system, as well make changes to actor.

### Proposal
* [ ] Remove `Actor` from the collision system and replace with a new facade `Collider`
     * [ ] Move collision type properties to `Collider` from Actor i.e `CollisionType`
* [ ] Rename confusing variables/terminology
     * [ ] `CollisionArea` -> `CollisionGeometry`
* [ ] Update documentation to match new terminology and usage

### PR DESCRIPTION
**Pull request by eonarheim**
===:clipboard: PR Checklist :clipboard:===

- [x] :pushpin: issue exists in github for these changes
- [x] :microscope: existing tests still pass
- [x] :see_no_evil: code conforms to the [style guide](https://github.com/excaliburjs/Excalibur/blob/master/STYLEGUIDE.md)
- [x] :triangular_ruler: new tests written and passing / old tests updated with new scenario(s)
- [x] :page_facing_up: changelog entry added (or not needed)

==================

Closes #1119

## Changes:

- New `ex.Collider` type which is the container for all collision related behavior and state. Actor is now extracted from collision.
- Added interface `Clonable<T>` to indicate if an object contains a clone method
- Added interface `Eventable<T>` to indicated if an object can emit and receive events
- `ex.Vector.scale` now also works with vector input
- `ex.BoundingBox.fromDimension(width: number, height: number)` can generate a bounding box from a width and height
- `ex.BoundingBox.translate(pos: Vector)` will create a new bounding box shifted by `pos`
- `ex.BoundingBox.scale(scale: Vector)` will create a new bounding box scaled by `scale`
- Added `isActor()` and `isCollider()` type guards

- Changed collision system to remove actor coupling, in addition `ex.Collider` is a new type that encapsulates all collision behavior. Use `ex.Actor.body.collider` to interact with collisions in Excalibur ([#1119](https://github.com/excaliburjs/Excalibur/issues/1119))

  - Add new `ex.Collider` type that is the housing for all collision related code
    - The source of truth for `ex.CollisionType` is now on collider, with a convenience getter on actor
    - The collision system now operates on `ex.Collider`'s not `ex.Actor`'s
  - `ex.CollisionType` has been moved to a separate file outside of `Actor`
    - CollisionType is switched to a string enum, style guide also updated
  - `ex.CollisionPair` now operates on a pair of `ex.Colliders`'s instead of `ex.Actors`'s
  - `ex.CollisionContact` now operates on a pair of `ex.Collider`'s instead of `ex.Actors`'s
  - `ex.Body` has been modified to house all the physical position/transform information
    - Integration has been moved from actor to `Body` as a physical concern
    - `useBoxCollision` has been renamed to `useBoxCollider`
    - `useCircleCollision` has been renamed to `useCircleCollider`
    - `usePolygonCollision` has been renamed to `usePolygonCollider`
    - `useEdgeCollision` has been renamed to `useEdgeCollider`
  - Renamed `ex.CollisionArea` to `ex.CollisionShape`
    - `ex.CircleArea` has been renamed to `ex.Circle`
    - `ex.PolygonArea` has been renamed to `ex.ConvexPolygon`
    - `ex.EdgeArea` has been renamed to `ex.Edge`
  - Renamed `getWidth()` & `setWidth()` to property `width`
    - Actor and BoundingBox are affected
  - Renamed `getHeight()` & `setHeight()` to property `height`
    - Actor and BoundingBox are affected
  - Renamed `getCenter()` to the property `center`
    - Actor, BoundingBox, and Cell are affected
  - Renamed `getBounds()` to the property `bounds`
    - Actor, Collider, and Shapes are affected
  - Renamed `getRelativeBounds()` to the property `localBounds`
    - Actor, Collider, and Shapes are affected
  - Renamed `moi()` to the property `inertia` standing for moment of inertia
  - Renamed `restition` to the property `bounciness`
  - Moved `collisionType` to `Actor.body.collider.type`
  - Moved `Actor.integrate` to `Actor.body.integrate`
- Removed `NaiveCollisionBroadphase` as it was no longer used
- Renamed methods and properties will be available until `v0.24.0`
- Deprecated collision attributes on actor, use `Actor.body.collider`

  - `Actor.x` & `Actor.y` will be removed in `v0.24.0` use `Actor.pos.x` & `Actor.pos.y`
  - `Actor.collisionArea` will be removed in `v0.24.0` use `Actor.body.collider.shape`
  - `Actor.getLeft()`, `Actor.getRight()`, `Actor.getTop()`, and `Actor.getBottom` are deprecated
    - Use `Actor.body.collider.bounds.(left|right|top|bottom)`
  - `Actor.getGeometry()` and `Actor.getRelativeGeometry()` are removed, use `Collider`
  - Collision related properties on Actor moved to `Collider`, use`Actor.body.collider`
    - `Actor.torque`
    - `Actor.mass`
    - `Actor.moi`
    - `Actor.friction`
    - `Actor.restition`
  - Collision related methods on Actor moved to `Collider`, use`Actor.body.collider` or `Actor.body.collider.bounds`
    - `Actor.getSideFromIntersect(intersect)` -> `BoundingBox.sideFromIntersection`
    - `Actor.collidesWithSide(actor)` -> `Actor.body.collider.bounds.intersectWithSide`
    - `Actor.collides(actor)` -> `Actor.body.collider.bounds.intersect`

### CODE REVIEW

File: src/engine/Collision/Collider.ts [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+  }
+
+  /**
+   * Gets or sets the current collision type of this collider. By
+   * default it is ([[CollisionType.PreventCollision]]).
+   */
+  public collisionType: CollisionType = CollisionType.PreventCollision;
+
+  public get shape(): CollisionGeometry {
+    return this._collisionArea;
```
Comment (eonarheim) [inline src/engine/Collision/Collider.ts]: rename to geometry

File: src/engine/Collision/CollisionContact.ts [hunk 2]
```diff
@@ -19,11 +18,11 @@ export class CollisionContact {
   /**
    * The first rigid body in the collision
```
Comment (eonarheim) [inline src/engine/Collision/CollisionContact.ts]: Update comments to match

File: src/engine/Collision/CollisionContact.ts [hunk 3]
```diff
@@ -19,11 +18,11 @@ export class CollisionContact {
...
    */
-  bodyA: CollisionArea;
+  colliderA: Collider;
   /**
    * The second rigid body in the collision
    */
-  bodyB: CollisionArea;
+  colliderB: Collider;
   /**
    * The minimum translation vector to resolve penetration, pointing away from bodyA
```
Comment (eonarheim) [inline src/engine/Collision/CollisionContact.ts]: Update comment to `colliderA`

File: src/engine/Integrator.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
+import { Vector } from './Algebra';
```
Comment (eonarheim) [inline src/engine/Integrator.ts]: Delete this file

File: src/engine/Physics.ts [hunk 5]
```diff
@@ -28,7 +28,7 @@ export enum BroadphaseStrategy {
 /**
  * Possible numerical integrators for position and velocity
  */
-export enum Integrator {
+export enum IntegratorType {
```
Comment (eonarheim) [inline src/engine/Physics.ts]: Revert this file

File: src/engine/Collision/Collider.ts [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+  return !!x && x instanceof Collider;
+}
+
+// Describes material properties like shape, bounds, friction of the physics object
+export class Collider implements Eventable {
+  private _collisionArea: CollisionGeometry;
+
+  constructor(private _actor: Actor, private _body: Body) {}
+
+  emit(eventName: string, event?: GameEvent<any>): void {
```
Comment (eonarheim) [inline src/engine/Collision/Collider.ts]: I'm not super pleased with this, maybe we don't this long term? Perhaps come up with some automatic bubbling strategy?

File: src/engine/Collision/Body.ts [hunk 7]
```diff
@@ -107,134 +122,168 @@ export class Body {
...
-  /**
-   * Returns the body's [[BoundingBox]] calculated for this instant in world space.
-   */
-  public getBounds(): BoundingBox {
-    if (Physics.collisionResolutionStrategy === CollisionResolutionStrategy.Box) {
-      return this.actor.getBounds();
-    } else {
-      return this.collisionArea.getBounds();
-    }
+  public taintCollisionGeometry() {
```
Comment (eonarheim) [inline src/engine/Collision/Body.ts]: Rename to `dirtyCollisionGeometry()`

Comment (kamranayub) [reply src/engine/Collision/Body.ts]: `markCollisionGeometryAsDirty()` ?

File: src/engine/Collision/Collider.ts (hunk 6 above)
Comment (eonarheim) [reply src/engine/Collision/Collider.ts]: Put these events into a array/dictionary and lookup the type and redispatch

File: src/engine/Collision/Collider.ts [hunk 8]
```diff
@@ -0,0 +1,176 @@
...
+   */
+  public get id(): number {
+    return this._actor.id;
+  }
+
+  /**
+   * Gets or sets the current collision type of this collider. By
+   * default it is ([[CollisionType.PreventCollision]]).
+   */
+  public collisionType: CollisionType = CollisionType.PreventCollision;
```
Comment (eonarheim) [inline src/engine/Collision/Collider.ts]: Should this just be `type`?

File: src/engine/Collision/Collider.ts [hunk 9]
```diff
@@ -0,0 +1,176 @@
...
+   * Get the shape of the collider as a [[CollisionGeometry]]
+   */
+  public get shape(): CollisionGeometry {
+    return this._geometry;
+  }
+
+  /**
+   * Set the shape of the collider as a [[CollisionGeometry]]
+   */
+  public set shape(shape: CollisionGeometry) {
```
Comment (eonarheim) [inline src/engine/Collision/Collider.ts]: Should `CollisionGeometry` be `CollisionShape`? This is more clear

File: src/engine/Collision/Collider.ts [hunk 10]
```diff
@@ -0,0 +1,176 @@
...
+
+  /**
+   * The current mass of the actor, mass can be thought of as the resistance to acceleration.
+   */
+  public mass: number = 1.0;
+
+  /**
+   * The current moment of inertia, moi can be thought of as the resistance to rotation.
+   */
+  public moi: number = 1000;
```
Comment (eonarheim) [inline src/engine/Collision/Collider.ts]: Should we just call this `inertia`?

File: src/engine/Collision/Collider.ts [hunk 11]
```diff
@@ -0,0 +1,176 @@
...
+
+  /**
+   * The coefficient of restitution of this actor, represents the amount of energy preserved after collision
+   */
+  public restitution: number = 0.2;
+
+  /**
+   * Returns the body's [[BoundingBox]] calculated for this instant in world space.
+   */
+  public getBounds(): BoundingBox {
```
Comment (eonarheim) [inline src/engine/Collision/Collider.ts]: Should `Collider` be the source of truth of for bounds for an actor?

File: src/engine/Collision/Collider.ts [hunk 12]
```diff
@@ -0,0 +1,176 @@
...
+      return this._actor.getBounds();
+    } else {
+      return this.shape.getBounds();
+    }
+  }
+
+  /**
+   * Returns the actor's [[BoundingBox]] relative to the actors position.
+   */
+  public getRelativeBounds(): BoundingBox {
```
Comment (eonarheim) [inline src/engine/Collision/Collider.ts]: Relative vs. world coordinates? This is confusing

File: src/engine/Collision/Collider.ts (hunk 8 above)
Comment (eonarheim) [reply src/engine/Collision/Collider.ts]: Is this really a `type` of behavior, or a `mode` of operation?

Comment (eonarheim) [reply src/engine/Collision/Collider.ts]: Group consensus is `type`, for ECS collider should be opt-in active/fixed

File: src/engine/Collision/CollisionGeometry.ts [hunk 13]
```diff
@@ -3,22 +3,29 @@ import { CollisionContact } from './CollisionContact';
...
 import { Vector, Projection, Ray } from '../Algebra';
+import { Collider } from './Collider';
 
 /**
- * A collision area is a region of space that can detect when other collision areas intersect
+ * A collision area specifies the geometry that can detect when other collision areas intersect
  * for the purposes of colliding 2 objects in excalibur.
  */
-export interface CollisionArea {
+export interface CollisionGeometry {
```
Comment (eonarheim) [inline src/engine/Collision/CollisionGeometry.ts]: Group consensus, rename to `CollisionShape`

File: src/engine/Collision/ConvexPolygon.ts [hunk 14]
```diff
@@ -1,38 +1,52 @@
...
+import { Vector, Line, Ray, Projection } from '../Algebra';
+import { Collider } from './Collider';
 
-export interface PolygonAreaOptions {
+export interface ConvexPolygonOptions {
   pos?: Vector;
   points?: Vector[];
   clockwiseWinding?: boolean;
+  collider?: Collider;
+  // @obsolete Will be removed in v0.24.0 please use [[collider]] to set and retrieve body information
```
Comment (eonarheim) [inline src/engine/Collision/ConvexPolygon.ts]: Needs to be jsdoc

File: src/engine/Collision/ConvexPolygon.ts [hunk 15]
```diff
@@ -1,38 +1,52 @@
...
 
 /**
  * Polygon collision area for detecting collisions for actors, or independently
  */
-export class PolygonArea implements CollisionArea {
+export class ConvexPolygon implements CollisionGeometry {
   public pos: Vector;
   public points: Vector[];
+
+  // @obsolete Will be removed in v0.24.0 please use [[collider]] to set and retrieve body information
```
Comment (eonarheim) [inline src/engine/Collision/ConvexPolygon.ts]: Needs to be jsdoc

File: src/engine/Collision/ConvexPolygon.ts [hunk 16]
```diff
@@ -1,38 +1,52 @@
...
 
-  constructor(options: PolygonAreaOptions) {
+  constructor(options: ConvexPolygonOptions) {
     this.pos = options.pos || Vector.Zero;
     const winding = !!options.clockwiseWinding;
     this.points = (winding ? options.points.reverse() : options.points) || [];
-    this.body = options.body || null;
+    this.collider = this.collider = options.collider || null;
+
+    // @obsolete Remove next release in v0.24.0, code exists for backwards compat
```
Comment (eonarheim) [inline src/engine/Collision/ConvexPolygon.ts]: Maybe make this a TODO

File: src/engine/Collision/NaiveCollisionBroadphase.ts [hunk 17]
```diff
@@ -1,9 +1,10 @@
 import { Physics } from './../Physics';
```
Comment (eonarheim) [inline src/engine/Collision/NaiveCollisionBroadphase.ts]: Can we just delete this file????

Comment (eonarheim) [reply src/engine/Collision/NaiveCollisionBroadphase.ts]: Consensus dead code

File: src/engine/Deprecated.ts [hunk 18]
```diff
@@ -159,10 +159,10 @@ export type ICellArgs = CellArgs;
 export type ICollidable = Collidable;
 
 /**
- * @deprecated Use CollisionArea, ICollisionArea will be removed in v0.23.0
- * @obsolete Use CollisionArea, ICollisionArea will be removed in v0.23.0
+ * @deprecated Use CollisionGeometry, ICollisionArea will be removed in v0.23.0
```
Comment (eonarheim) [inline src/engine/Deprecated.ts]: We don't need `@deprecated` only `@obsolete`

File: src/engine/Actor.ts [hunk 19]
```diff
@@ -46,6 +45,13 @@ import * as Effects from './Drawing/SpriteEffects';
 import * as Util from './Util/Util';
 import * as Events from './Events';
 import { PointerEvents } from './Interfaces/PointerEvents';
+import { CollisionType } from './Collision/CollisionType';
+import { obsolete } from './Util/Decorators';
+import { Collider } from './Collision/Collider';
+
+export function isActor(x: any): x is Actor {
+  return !!x && x instanceof Actor;
```
Comment (kamranayub) [inline src/engine/Actor.ts]: Is the !!x redundant with instanceof?

**Comment by eonarheim** [comment]
Good question, probably...

File: src/engine/Collision/Collider.ts (hunk 12 above)
Comment (eonarheim) [reply src/engine/Collision/Collider.ts]: Agreed, this is confusing not feasible to change at the moment

File: src/engine/Actor.ts [hunk 20]
```diff
@@ -98,19 +104,29 @@ export class ActorImpl extends Class implements Actionable, Eventable, PointerEv
   public body: Body = new Body(this);
 
   /**
-   * Gets the collision area shape to use for collision possible options are [CircleArea|circles], [PolygonArea|polygons], and
-   * [EdgeArea|edges].
+   * Collider associated with this actor's body
+   */
+  public get collider(): Collider {
```
Comment (eonarheim) [inline src/engine/Actor.ts]: Remove convenience method? This introduces coupling preemptively.

File: src/engine/Actor.ts [hunk 21]
```diff
@@ -217,6 +228,20 @@ export class ActorImpl extends Class implements Actionable, Eventable, PointerEv
     this.body.acc.setTo(theAcc.x, theAcc.y);
   }
 
+  /**
+   * Sets the acceleration of the actor from the last frame. This does not include the global acc [[Physics.acc]].
+   */
+  public set oldAcc(theAcc: Vector) {
```
Comment (eonarheim) [inline src/engine/Actor.ts]: Deprecate the convenience methods off of actor that project from body

File: src/engine/Collision/Body.ts [hunk 22]
```diff
@@ -108,139 +124,184 @@ export class Body {
...
+  public useBoxCollider(center: Vector = Vector.Zero): Collider {
+    this.collider.shape = new ConvexPolygon({
+      collider: this.collider,
+      points: this._actor.getRelativeGeometry(),
       pos: center // position relative to actor
     });
 
     // in case of a nan moi, coalesce to a safe default
-    this.moi = this.collisionArea.getMomentOfInertia() || this.moi;
+    this.collider.inertia = this.collider.shape.getInertia() || this.collider.inertia;
```
Comment (eonarheim) [inline src/engine/Collision/Body.ts]: This seems odd, should this ever be NaN? Or should we have a better isNaN() check around this?

File: src/engine/Collision/Body.ts [hunk 23]
```diff
@@ -108,139 +124,184 @@ export class Body {
...
-    }
-
-    if (Physics.showBounds) {
-      this.getBounds().debugDraw(ctx, Color.Yellow);
-    }
-
-    if (Physics.showArea) {
-      this.collisionArea.debugDraw(ctx, Color.Green);
-    }
+  private _wireColliderEventsToActor() {
```
Comment (eonarheim) [inline src/engine/Collision/Body.ts]: Add TODO around this for ECS to remove, eventually events will stay local to the thing they are about

File: src/engine/Collision/BoundingBox.ts [hunk 24]
```diff
@@ -37,6 +18,26 @@ export class BoundingBox implements Collidable {
    */
   constructor(public left: number = 0, public top: number = 0, public right: number = 0, public bottom: number = 0) {}
 
+  public static sideFromIntersection(intersection: Vector): Side {
```
Comment (eonarheim) [inline src/engine/Collision/BoundingBox.ts]: Needs JSDoc comment explaining more

File: src/engine/Collision/BoundingBox.ts [hunk 25]
```diff
@@ -201,138 +202,156 @@ export class BoundingBox implements Collidable {
   }
 
   /**
-   * Test wether this bounding box collides with another returning,
+   * Test wether this bounding box intersects with another returning
    * the intersection vector that can be used to resolve the collision. If there
-   * is no collision null is returned.
+   * is no interesection null is returned.
```
Comment (eonarheim) [inline src/engine/Collision/BoundingBox.ts]: typo

File: src/engine/Collision/BoundingBox.ts [hunk 26]
```diff
@@ -201,138 +202,156 @@ export class BoundingBox implements Collidable {
...
    *
-   * @returns A Vector in the direction of the current BoundingBox
-   * @param collidable  Other collidable to test
+   * @returns A Vector in the direction of the current BoundingBox, this <- other
```
Comment (eonarheim) [inline src/engine/Collision/BoundingBox.ts]: Transpose these 2 lines

File: src/engine/Collision/Collider.ts [hunk 27]
```diff
@@ -0,0 +1,200 @@
...
+    return !this.entity.isKilled();
+  }
+
+  /**
+   * Collide 2 colliders and product a collision contact if there is a collision, null if none
+   *
+   * Collision vector is in the direction of the other collider. Away from this collider, this -> other.
+   * @param other
+   */
+  public collide(other: Collider): CollisionContact {
```
Comment (eonarheim) [inline src/engine/Collision/Collider.ts]: CollisionContact | null ?

File: src/engine/Collision/Collider.ts [hunk 28]
```diff
@@ -0,0 +1,200 @@
...
+
+  /**
+   * The coefficient of friction on this actor
+   */
+  public friction: number = 0.99;
+
+  /**
+   * The coefficient of restitution of this actor, represents the amount of energy preserved after collision
+   */
+  public restitution: number = 0.2;
```
Comment (eonarheim) [inline src/engine/Collision/Collider.ts]: Call this bounciness instead?

Comment (eonarheim) [reply src/engine/Collision/Collider.ts]: Maybe refer to this in the documentation as an aka "bounciness"

Comment (eonarheim) [reply src/engine/Collision/Collider.ts]: Or refer to the technical name in the docs

File: src/engine/Collision/Collider.ts [hunk 29]
```diff
@@ -0,0 +1,200 @@
...
+  }
+
+  /**
+   * Returns the actor's [[BoundingBox]] relative to the actors position.
+   */
+  public getRelativeBounds(): BoundingBox {
+    if (Physics.collisionResolutionStrategy === CollisionResolutionStrategy.Box) {
+      return this.entity.getRelativeBounds();
+    } else {
+      return this.entity.getRelativeBounds();
```
Comment (eonarheim) [inline src/engine/Collision/Collider.ts]:
```suggestion
      return this.shape.getRelativeBounds();
```

File: src/engine/Collision/CollisionShape.ts [hunk 30]
```diff
@@ -42,12 +54,12 @@ export interface CollisionArea {
...
    */
-  getMomentOfInertia(): number;
+  getInertia(): number;
 
-  // All new ICollisionAreas need to do the following
+  // All new CollisionShape need to do the following
   // Create a new collision function in the CollisionJumpTable against all the primitives
   // Currently there are 3 primitive collision areas 3! = 6 jump functions
-  collide(area: CollisionArea): CollisionContact;
+  collide(area: CollisionShape): CollisionContact;
```
Comment (eonarheim) [inline src/engine/Collision/CollisionShape.ts]:
```suggestion
  collide(shape: CollisionShape): CollisionContact;
```

File: src/engine/EventDispatcher.ts [hunk 31]
```diff
@@ -10,25 +10,30 @@ import { Eventable } from './Interfaces/Evented';
  *
  * [[include:Events.md]]
  */
-export class EventDispatcher implements Eventable {
+export class EventDispatcher<T = any> implements Eventable {
   private _handlers: { [key: string]: { (event: GameEvent<any>): void }[] } = {};
```
Comment (eonarheim) [inline src/engine/EventDispatcher.ts]:
```suggestion
  private _handlers: { [key: string]: { (event: GameEvent<T>): void }[] } = {};
```

File: src/engine/EventDispatcher.ts [hunk 32]
```diff
@@ -10,25 +10,30 @@ import { Eventable } from './Interfaces/Evented';
...
 
   /**
    * @param target  The object that will be the recipient of events from this event dispatcher
    */
-  constructor(target: any) {
+  constructor(target: T) {
     this._target = target;
   }
 
+  public clear() {
```
Comment (eonarheim) [inline src/engine/EventDispatcher.ts]: Needs JSDoc

File: src/engine/Events.ts [hunk 33]
```diff
@@ -158,11 +159,17 @@ export type pointerdragmove = 'pointerdragmove';
...
+export class GameEvent<T, U = T> {
   /**
    * Target object for this event.
    */
   public target: T;
+
+  /**
+   * Other target object for this event
+   */
+  public other: U;
```
Comment (eonarheim) [inline src/engine/Events.ts]: Make nullable

File: src/engine/Traits/TileMapCollisionDetection.ts [hunk 34]
```diff
@@ -18,7 +20,7 @@ export class TileMapCollisionDetection implements Trait {
           if (max-- < 0) {
             break;
           }
-          side = actor.getSideFromIntersect(intersectMap);
+          side = BoundingBox.sideFromIntersection(intersectMap);
```
Comment (eonarheim) [inline src/engine/Traits/TileMapCollisionDetection.ts]: Needs action verb, `get` or `determine` `sideFromIntersection`

File: src/engine/Collision/Collider.ts (hunk 29 above)
Comment (eonarheim) [reply src/engine/Collision/Collider.ts]: Actually for relative bounds, this makes sense

File: src/engine/Actor.ts [hunk 35]
```diff
@@ -329,17 +338,20 @@ export class ActorImpl extends Class implements Actionable, Eventable, PointerEv
   /**
    * Gets the coefficient of restitution of this actor, represents the amount of energy preserved after collision. Think of this
    * as bounciness.
+   * @obsolete ex.Actor.restitution will be removed in v0.24.0, use ex.Actor.body.collider.restitution
    */
+  @obsolete({ message: 'ex.Actor.restitution will be removed in v0.24.0', alternateMethod: 'ex.Actor.body.collider.restitution' })
```
Comment (kamranayub) [inline src/engine/Actor.ts]: alt method should be `collider.bounciness`

File: src/engine/Actor.ts [hunk 36]
```diff
@@ -329,17 +338,20 @@ export class ActorImpl extends Class implements Actionable, Eventable, PointerEv
...
   public get restitution() {
-    return this.body.collider.restitution;
+    return this.body.collider.bounciness;
   }
 
   /**
    * Sets the coefficient of restitution of this actor, represents the amount of energy preserved after collision. Think of this
    * as bounciness.
+   * @obsolete ex.Actor.restitution will be removed in v0.24.0, use ex.Actor.body.collider.restitution
```
Comment (kamranayub) [inline src/engine/Actor.ts]: alt method should be `collider.bounciness`

**Comment by eonarheim** [comment]
@excaliburjs/core-contributors I think we are ready

**Pull request merged**